include/EASTL/sort.h

```diff
@@ -797,10 +818,10 @@ namespace eastl
 			if(compare(*i, *first))
 			{
 				EASTL_VALIDATE_COMPARE(!compare(*first, *i)); // Validate that the compare function is sane.
-				const value_type temp(*i);
-				*i = *first;
+				value_type temp(eastl::forward<value_type>(*i));
```

Comment:
Still working on the unit test for this case... for some reason I can trigger the non-compare variant compile failure only via benchmarks but not unit tests.... investigating root cause...

Once I figure out why benchmarks and unit-tests invoke different code paths I'll add a new unit tests.